### ISSUE TITLE
'Braintree/Braintree-Swift.h' file not found in React Native project using CocoaPods

### ISSUE THREAD

**Issue by jbatra-umeey** [braintree/braintree_ios#671]
**Integration Details**

 - Braintree SDK Version: 5.3.0
 - Environment: : Sandbox
 - iOS Version: : iOS 13.4.1
 - Device: [e.g. iPhone 12] : iPhone 7
 - Integration type & version: Cocoapods 1.10.1

**Describe the bug**

When updating to 5.3.0 we get an error during the compilation of the header file 'Braintree/Braintree-Swift.h' . The current version we are using is 4.37.1
In continuation to issue https://github.com/braintree/braintree_ios/issues/651

Podfile :

```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '12.0'
use_frameworks!
inhibit_all_warnings!
target 'demoReactNativeApp' do
  config = use_native_modules!

pod 'Braintree'
pod 'Braintree/Venmo'

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => false
  )

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  use_flipper!()

  post_install do |installer|
    react_native_post_install(installer)
  end

end
```

**To Reproduce**
There are no explicit steps to reproduce, other than try to update the version

**Expected behavior**
Success build with the latest version

**Screenshots**
<img width="1802" alt="Screenshot 2021-05-05 at 1 34 46 PM" src="https://user-images.githubusercontent.com/75033678/117113860-b48a3700-ada8-11eb-9fef-573c41deb00d.png">

In the project directory, I can see this file present. Any changes needed in the main project's build settings?

<img width="1352" alt="Screenshot 2021-05-05 at 4 05 34 PM" src="https://user-images.githubusercontent.com/75033678/117129237-0ee0c300-adbc-11eb-8212-6680d31a1609.png">

As a workaround can we skip `PayPalDataCollector ` from cocoa pods?

**Comment by sestevens** [comment]
> As a workaround can we skip PayPalDataCollector from cocoa pods?

This depends on which payment methods you're using. If you're using PayPal, which is included by default with `pod 'Braintree'`, then `PayPalDataCollector` is required. You can take a look at all of the dependencies in [`Braintree.podspec`](https://github.com/braintree/braintree_ios/blob/master/Braintree.podspec).

As mentioned in #651, we don't officially support React Native. However, if you have any suggestions or ideas about how to fix this issue, we're more than happy to review a PR.

**Comment by jbatra-umeey** [comment]
@sestevens Thanks for the prompt revert. Any tips to debug it?

Few workarounds I tried : 

- https://developer.apple.com/forums/thread/651941
- Manually added the header search path for the umbrella header

I suppose we should also have a look at Braintree. modulemap

```
module Braintree {
  umbrella header "Braintree-Swift.h"

  export *
  module * { export * }
}

```

**Comment by sestevens** [comment]
I found a [similar issue](https://github.com/CocoaPods/CocoaPods/issues/6351) on CocoaPods, but it looks like the solution there was to use angle brackets `< >` on the import instead of quotes, which we're already doing.

I also see a similar issue [here](https://github.com/prscX/react-native-notification-banner/issues/13#issuecomment-562114163), and the solution was to use `@import` instead of `#import`. That might be worth a try, as long as it doesn't break anything for integrations that aren't using React Native.

**Comment by jbatra-umeey** [comment]
Ok, I managed to fix the umbrella header  issue in BTPayPalDriver.m by replacing `#import <Braintree/Braintree-Swift.h>` with
`#import "Braintree-Swift.h"`

 I am getting the below issue how :
 
```
Undefined symbols for architecture arm64:
  "__swift_FORCE_LOAD_$_swiftWebKit", referenced from:
      __swift_FORCE_LOAD_$_swiftWebKit_$_Braintree in libBraintree.a(PPDataCollector.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftWebKit_$_Braintree)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by sestevens** [comment]
@jbatra-umeey Could you post a link to the React Native sample app you're using?

**Comment by jbatra-umeey** [comment]
@sestevens  I kept https://github.com/wgltony/react-native-braintree-dropin-ui as the base and customized based on my requirements. For now, maybe we could use this one for testing

**Comment by jbatra-umeey** [comment]
@sestevens could you share an update on the suggested changes?. Also, please let me know if you need any help in compiling the  react native sample,

**Comment by sestevens** [comment]
Hi @jbatra-umeey, we haven't had a chance to reproduce this yet, but we're planning to take a look at the sample app you linked to. I want to be careful about using `#import "Braintree-Swift.h"` because we've had issues with `"`-style imports before. But we'll try it out when we get a chance! 

We don't have an exact timeline for when we'll have a chance to tackle this, but I'll keep you posted on any progress.

**Comment by jbatra-umeey** [comment]
@sestevens I tried migrating to SPM, the source files build fine. However, when I try to import the files into a pod dependency, it throws a compilation error for header files since SPM is part of its main target.
Do you have any tips to work with such arrangements?. By any chance, the header path needs to be set manually?

**Comment by jbatra-umeey** [comment]
@sestevens is there any sample to compile the code using source files?

**Comment by sestevens** [comment]
@jbatra-umeey Here's a [demo app](https://github.com/braintree/braintree_ios/tree/master/Demo) that's included in this repo. Also, we tried changing the import to `#import "Braintree-Swift.h"`, but it's causing `pod lib lint` to fail with an error about not being able to find that file. We'll keep investigating and let you know if we come up with a solution.

**Comment by jbatra-umeey** [comment]
@sestevens would you recommend building the pod from local file path of SPM ?
 
For example, in my case, the dependency of my development pod is on :


  ```
s.dependency    'BraintreeDropIn'
  s.dependency    'Braintree/DataCollector'
  s.dependency    'Braintree/Apple-Pay'
  s.dependency    'Braintree/Venmo'

```

In podfile:

pod 'Braintree', :path => ' path to brain tree-master/'

**Comment by scannillo** [comment]
👋  Hello @jbatra-umeey - I am able to get `#import "Braintree-Swift.h"` working for a React-Native app. After that, I am also able to replicate the `swiftWebKit` type issues you were seeing.

Question - is the native iOS portion of your React-Native app written in pure Objective-C? 

If so, I was able to resolve the `swiftWebKit` type issue by adding an empty Swift file into my application, and letting XCode properly generate and link a bridging header. ([SO post](https://stackoverflow.com/questions/50096025/it-gives-errors-when-using-swift-static-library-with-objective-c-project/50495316#50495316) for reference). This removed the need to manually include `libSwiftWebKit` & other similar frameworks. 

Please let me know if that gets things working for you.

**Comment by jbatra-umeey** [comment]
@scannillo  thanks for the quick update. I will try it out `react-native` branch on my end and share the update
Yes, the code is written in Objective-C
I believe the changes you pointed out w.r.t `swiftWebKit`should work as per [SO post](https://stackoverflow.com/questions/50096025/it-gives-errors-when-using-swift-static-library-with-objective-c-project/50495316#50495316)

**Comment by scannillo** [comment]
👋  Hello. We have released a fix for this in [Braintree iOS v5.3.2](https://github.com/braintree/braintree_ios/releases/tag/5.3.2).
